src/wp-includes/class-wp-theme-json-resolver.php

```diff
@@ -122,75 +140,243 @@ public static function get_core_data() {
+		}
+
+		return $user_cpt;
+	}
+
+	/**
+	 * Returns the user's origin config.
+	 *
+	 * @since 5.9.0
+	 *
+	 * @return WP_Theme_JSON Entity that holds user data.
+	 */
+	public static function get_user_data() {
+		if ( null !== self::$user ) {
+			return self::$user;
+		}
+
+		$config   = array();
+		$user_cpt = self::get_user_data_from_custom_post_type();
+		if ( array_key_exists( 'post_content', $user_cpt ) ) {
+			$decoded_data = json_decode( $user_cpt['post_content'], true );
+
+			$json_decoding_error = json_last_error();
+			if ( JSON_ERROR_NONE !== $json_decoding_error ) {
+				trigger_error( 'Error when decoding a theme.json schema for user data. ' . json_last_error_msg() );
+				return new WP_Theme_JSON( $config, 'user' );
+			}
+
+			// Very important to verify if the flag isGlobalStylesUserThemeJSON is true.
+			// If is not true the content was not escaped and is not safe.
+			if (
+				is_array( $decoded_data ) &&
+				isset( $decoded_data['isGlobalStylesUserThemeJSON'] ) &&
+				$decoded_data['isGlobalStylesUserThemeJSON']
+			) {
+				unset( $decoded_data['isGlobalStylesUserThemeJSON'] );
+				$config = $decoded_data;
+			}
+		}
+		self::$user = new WP_Theme_JSON( $config, 'user' );
+
+		return self::$user;
+	}
+
+	/**
+	 * There are three sources of data (origins) for a site:
+	 * core, theme, and user. The user's has higher priority
+	 * than the theme's, and the theme's higher than core's.
+	 *
+	 * Unlike the getters {@link get_core_data},
+	 * {@link get_theme_data}, and {@link get_user_data},
+	 * this method returns data after it has been merged
+	 * with the previous origins. This means that if the same piece of data
+	 * is declared in different origins (user, theme, and core),
+	 * the last origin overrides the previous.
+	 *
+	 * For example, if the user has set a background color
+	 * for the paragraph block, and the theme has done it as well,
+	 * the user preference wins.
 	 *
 	 * @since 5.8.0
+	 * @since 5.9.0 Add user data and change the arguments.
 	 *
-	 * @param array $settings Optional. Existing block editor settings. Default empty array.
+	 * @param string $origin To what level should we merge data.
+	 *                       Valid values are 'theme' or 'user'.
+	 *                       Default is 'user'.
 	 * @return WP_Theme_JSON
 	 */
-	public static function get_merged_data( $settings = array() ) {
-		$theme_support_data = WP_Theme_JSON::get_from_editor_settings( $settings );
-
+	public static function get_merged_data( $origin = 'user' ) {
 		$result = new WP_Theme_JSON();
 		$result->merge( self::get_core_data() );
-		$result->merge( self::get_theme_data( $theme_support_data ) );
+		$result->merge( self::get_theme_data() );
+
+		if ( 'user' === $origin ) {
+			$result->merge( self::get_user_data() );
+		}
 
 		return $result;
 	}
 
+	/**
+	 * Registers a Custom Post Type to store the user's origin config.
+	 *
+	 * @since 5.9.0
+	 */
+	public static function register_user_custom_post_type() {
```

Comment:
This function is not being called so the post type is not registered. I guess with the move to the core we can register the post type with the other core post types and remove this function?

Reply:
Hi @oandregal, I moved the registration to be consistent with other post types and also updated the taxonomy in a PR just for that https://github.com/WordPress/wordpress-develop/pull/1819. We can remove this function as it is not required.

Reply:
:+1: